irispy/sji.py

```diff
@@ -0,0 +1,188 @@
+                                    bitrate=64000)
+    if not writer.isAvailable():
+        path = input('Set FFMpeg path: (ex. /usr/local/bin/ffmpeg)')
+        plt.rcParams['animation.ffmpeg_path'] = path
+    writer = animation.FFMpegWriter(fps=fps, metadata=dict(artist='SunPy'),
+                                    bitrate=64000)
+    if not writer.isAvailable():
+        print('Cannot find available FFMpeg Writer')
+        return
+    plt.rcParams['axes.facecolor'] = 'black'
+    plt.rcParams['xtick.direction'] = 'out'
+    plt.rcParams['ytick.direction'] = 'out'
+    ax = plt.axes(zorder=0)
+
+    ani = mc.plot(ax)
+    ani.save(outputfile, writer=writer)
+    print('Successfully saved to ' + outputfile)
+
+
+def dustbuster(filelist):
+    """
+       Read SJI fits files and return Inpaint-corrected fits files.
+       **Warning** Will clobber input file.
+
+       Parameters
+       ----------
+       filelist: string or list
+           File to read, if single file string detected, will
+           create a list of length 1.
+
+
+       Returns
+       -------
+       NoneType
+           Inpaint-corrected fits to clobber input file
+       """
+
+    if type(filelist) == str:  # convert single file to list
+        file_list = [filelist]
+    nfile = 0
+    for file in filelist:
+        nfile += 1
+        image_header = fits.getheader(file)
+        image_orig = fits.getdata(file)
+        nx = image_header.get('NRASTERP')
+        print(image_header)
+        # TODO mapcube input
+
+        img_shape = image_orig.shape
+        ndx = img_shape[0]
+
+        image_result = []
+        counter = 0
+        firstpos = range(ndx)[0::nx]
+
+        for i in range(0, ndx):
+            #  Create mask with values < 1, excluding frame (-200)
+            mask = np.zeros(image_orig[i].shape)
+            mask[np.where(image_orig[i] < 1)] = 1
+            mask[np.where(image_orig[i] == -200)] = 0
+
+            image_fix = image_orig[i].copy()
+            if nx <= 50:     # sparse/coarse raster
+                skip = 1
+            elif nx > 50:    # dense raster
+                skip = 3
+                secpos = range(ndx)[1::nx]
+                thirdpos = range(ndx)[2::nx]
+
+            if (i in firstpos) or (i in secpos) or (i in thirdpos):
+                image_inpaint = image_orig[i + skip].copy()  # grab next frame
+            else:
+                image_inpaint = image_orig[i - skip].copy()  # grab prev frame
+
+            # Inpaint mask onto image
+            for layer in range(image_fix.shape[-1]):
+                image_fix[np.where(mask)] = image_inpaint[np.where(mask)]
+
+            image_result.append(image_fix)  # add corrected image to new list
+            counter += 1
+            print(str(counter) + ' of ' + str(ndx))
+
+        #  overwrite old file
+        print(len(image_result))
+        outputfile = file
+        fits.writeto(outputfile, image_result, header=image_header,
```

Comment:
I suggest you modify this so it can optionally save to new files rather than overwriting.

Also I would make it return the list of files that it has modified or created.